---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# RPracticePackage

<!-- badges: start -->
[![GitHub issues](https://img.shields.io/github/issues/srp33/annotated_breast_cancer_ge_data)](https://github.com/srp33/annotated_breast_cancer_ge_data/issues)
[![GitHub pulls](https://img.shields.io/github/issues-pr/srp33/annotated_breast_cancer_ge_data)](https://github.com/srp33/annotated_breast_cancer_ge_data/pulls)
<!-- badges: end -->

The goal of `AnnotatedBCGEData` is to provide users with easy access to 100+ tidied breast cancer gene expression data sets from the Gene Expression Omnibus.

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `AnnotatedBCGEData` from [Bioconductor](http://bioconductor.org/) using the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("AnnotatedBCGEData")
```

This code installs BiocManager if it is not already installed. Running the if statement is required to install the package as it is dependent on BiocManager to load. 

Download the development version from [GitHub](https://github.com/srp33/annotated_breast_cancer_ge_data) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("srp33/annotated_breast_cancer_ge_data")
```

## Data Info

The data used in this package was accessed through Open Science Framework (OSF). This database allowed us to use data that was already neat and ready to be analyzed. However, the original data came from Gene Expression Omnibus (GEO). The name of every item in identifier_list is the data set name in GEO. Learn more about the data at the following links:

 - GSE41197: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41197 
 - GSE10797: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10797 
 - GSE59772: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59772 

The package itself includes a metadata file as required by Bioconductor. This file is a table that lists the name of each object and more information about where it came from and how it was accessed. See inst/extdata/metadata.csv. When the package is available in ExperimentHub, using the query() function will show available objects. 

```{r load_metadata}
#library(ExperimentHub)
#eh = ExperimentHub()
#query(eh, AnnotatedBCGEData)

# this code throws errors because the package is not currently available in ExperimentHub

```


## Examples

The following is the code required to access the objects within the package, as well as additional packages we used to analyze the data. The SummarizedExperiment package is required for both building the package and accessing the data. 
```{r load_packages, message=FALSE, warning=FALSE}
library(AnnotatedBCGEData)
library(SummarizedExperiment)
library(tidyverse)
```

Downloading this package does not download the data sets onto the user's machine. This package contains 2 code objects the user needs to create and access the SummarizedExperiment object for each data set. First is a named list object called identifiers. The second is a function called makeObject. identifiers contains the unique URL identifiers for the expression data file and the metadata file. Calling each data set in the named list returns a vector with these values. These values are passed to makeObject, which returns the SummarizedExperiment. To see the contents of the named list, run the following code. Note that we have suppressed the output of the code for readability. Additionally, this code is not required in order to create the SummarizedExperiment objects.

```{r load_identifiers, echo=TRUE, results='hide'}
data('identifiers')
identifiers

```

To load the SummarizedExperiment object for a data set included in AnnotatedBCGEData, the user chooses the name of the data set from the identifiers named list object and passes it to the function makeObject. This function downloads the data and creates the SummarizedExperiment object that is accessible to the user.  <br>
<br>
The SummarizedExperiment object for each data set includes 3 matrices. The first, which we load as expression_data, is a matrix with the samples as columns and genes as rows. The Ensembl gene ID is used as the row names. The data in the actual matrix is the expression level of the gene for the sample. <br>
<br>
The second matrix we load as sample_metadata. This is information about the samples themselves, such as demographics for the patient the sample came from. The samples are rows and the information is in the columns. <br>
<br>
The third matrix we load as feature_data. This is the information about the genes that were tested for each sample. This includes information like the chromosome the gene is located on, the Entrez gene ID, and the gene name. The genes are the rows and the information about the samples is in the columns.

```{r example, eval = requireNamespace('AnnotatedBCGEData')}
GSE41197_SE = makeObject(identifiers$GSE41197)
expression_data = assay(GSE41197_SE)
sample_metadata = colData(GSE41197_SE)
feature_data = rowData(GSE41197_SE)
```

This is a simple example of using a built-in function to get a general idea of the data. The summary() function includes data points for each sample, like the minimum or maximum expression levels across all the genes. Generally, this isn't particularly relevant, but does give an idea of how many samples are in the matrix.

```{r experiment_summary}
gene_expression_values = assay(GSE41197_SE)
summary(gene_expression_values)
```

In the following example, we first access the matrix of gene expression values. To make it easier to analyze, we convert it into a tibble and select the first 10 rows. We then used ggplot() to create a bar plot showing the gene expression levels for one sample across 10 genes.

```{r plot_example}
GSE41197 = makeObject(identifiers$GSE41197)
gene_expression_values = assay(GSE41197)

exp_tib = as_tibble(gene_expression_values, rownames='Ensembl_Gene_ID')[1:10,] %>%
    ggplot(aes(x=Ensembl_Gene_ID, y=GSM1010328)) +
    geom_col() +
    labs(x='Ensembl Gene ID',y='Sample ID GSM1010328') +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))

exp_tib
```


Here, we first accessed the matrices for 3 data sets included in the RPracticePackage. 

```{r plot_example_2}
GSE10797_SE = makeObject(identifiers$GSE10797)
GSE41197_SE = makeObject(identifiers$GSE41197)
GSE59772_SE = makeObject(identifiers$GSE59772)

GSE10197 = assay(GSE10797_SE)
GSE41197 = assay(GSE41197_SE)
GSE59772 = assay(GSE59772_SE)
```

Next, we converted each to a tibble and selected one row, representing one gene that we knew each data set had in common. We rotated the tibbles to have one column (the gene) and many rows (the sample). Then, we added the 'group' column, which was necessary for the function used in the next step.

```{r plot_example_2_part_2}
GSE10197_gene = GSE10197 %>%
    as_tibble(rownames='Ensembl_Gene_ID') %>%
    filter(Ensembl_Gene_ID == "ENSG00000171428") %>%
    pivot_longer(cols=-Ensembl_Gene_ID,names_to='Sample_ID',values_to='Expression_Level')%>%
    mutate(group = 'GSE10197')
GSE41197_gene = GSE41197 %>%
    as_tibble(rownames='Ensembl_Gene_ID') %>%
    filter(Ensembl_Gene_ID == "ENSG00000171428") %>%
    pivot_longer(cols=-Ensembl_Gene_ID,names_to='Sample_ID',values_to='Expression_Level')%>%
    mutate(group = 'GSE41197')
GSE59772_gene = GSE59772 %>%
    as_tibble(rownames='Ensembl_Gene_ID') %>%
    filter(Ensembl_Gene_ID == "ENSG00000171428") %>%
    pivot_longer(cols=-Ensembl_Gene_ID,names_to='Sample_ID',values_to='Expression_Level')%>%
    mutate(group = 'GSE59772')
```

We then joined the 3 tibbles into one tibble with the bind_rows() function. We now have one tibble with one column (the gene) and many rows (the samples). <br>
<br>
Finally, we used ggplot() to visualize the expression levels for the gene across the three data sets.

```{r plot_example_2_part_3}
combined_samples = bind_rows(GSE10197_gene, GSE41197_gene, GSE59772_gene)

ggplot(combined_samples, aes(x = group, y = Expression_Level, fill = group)) +
    geom_boxplot() +
    labs(x = "Dataset", y = "Expression Level") +
    theme_bw()
```

In this example, we combine using both the metadata and the expression data to create a heat map of gene expression levels.<br>
First, we load the data sets, turn them into tibbles, and extract both the metadata and expression data. We then filter the
expression to only include the samples that are identified as stroma cells. Finally, we create two vectors with all the gene names in each tibble to set them up for comparison. 

``` {r heat_map}
GSE59772 = makeObject(identifiers$GSE59772)
GSE59772_metadata = colData(GSE59772) %>%
    as_tibble(rownames = 'Sample_ID')

GSE59772_stroma_samples = filter(GSE59772_metadata, tissue == 'Stroma') %>%
    pull(Sample_ID)

GSE59772_gene_names = assay(GSE59772) %>%
    as_tibble(rownames="Ensembl_Gene_ID") %>%
    pull(Ensembl_Gene_ID)


GSE10797 = makeObject(identifiers$GSE10797)
GSE10797_metadata = colData(GSE10797) %>%
    as_tibble(rownames = 'Sample_ID')

GSE10797_gene_names = assay(GSE10797) %>%
    as_tibble(rownames="Ensembl_Gene_ID") %>%
    pull(Ensembl_Gene_ID)

GSE10797_stroma_samples = filter(GSE10797_metadata, tissue_source == 'stromal cells from breast cancer patient')%>%
    pull(Sample_ID)
```

Next, we create a vector that has only the names of the genes the data sets have in common. We use this vector to filter the expression tibbles again to only include rows with genes in the vector. 

``` {r heat_map_2}
common_genes = c()
for (gene in GSE59772_gene_names) {
    if (gene %in% GSE10797_gene_names) {
        common_genes = c(common_genes, gene)
    }
}

GSE59772_stroma_expressions = assay(GSE59772) %>%
    as_tibble(rownames= 'Ensembl_Gene_ID') %>%
    select(Ensembl_Gene_ID, all_of(GSE59772_stroma_samples)) %>%
    filter(Ensembl_Gene_ID %in% common_genes)
    
GSE10797_stroma_expressions = assay(GSE10797) %>%
    as_tibble(rownames = 'Ensembl_Gene_ID') %>%
    select(Ensembl_Gene_ID, all_of(GSE10797_stroma_samples)) %>% 
    filter(Ensembl_Gene_ID %in% common_genes)

```

Finally, we combine the data into one tibble called combined_data. We make a new row called variance that calculates the variance across the 5 samples. We select the top 10 rows, those with greatest variance. We then use ggplot to create a heat map of the genes with the top variance, displayed below. 

``` {r heat_map_3}
combined_data = full_join(GSE10797_stroma_expressions, GSE59772_stroma_expressions)
top_variance_genes = combined_data %>%
    rowwise() %>%
    mutate(variance = var(c_across(starts_with('GSM')))) %>%
    ungroup() %>%
    arrange(desc(variance)) %>%
    slice_head(n=10)

top_variance_map = top_variance_genes %>%
    select(-variance) %>%
    pivot_longer(cols = starts_with('GSM'), names_to = 'Sample_ID', values_to = 'Expression') %>%
    ggplot(aes(x = Sample_ID, y = Ensembl_Gene_ID, fill = Expression)) +
    geom_tile()+
    labs(x = 'Sample', y = 'Gene', fill = 'Expression', title = 'Top 10 Genes in Stroma Samples with Highest Variance') + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(hjust = 0.5, face = "bold"))
top_variance_map


```

In this final example, we use aspects of the feature data along with the expression data. <br> <br>

First, we get the feature data and convert it to a tibble. We then filter this data for each data set to only include genes on chromosome 16. Similar to the last example, we also make a vector with the gene names the data sets have in common. 
``` {r chr16_boxplot}
GSE59772 = makeObject(identifiers$GSE59772)
GSE59772_featuredata = rowData(GSE59772) %>%
    as_tibble(rownames = 'Ensembl_Gene_ID')

GSE59772_chr16 = GSE59772_featuredata %>%
    filter(Chromosome == '16') %>%
    pull(Ensembl_Gene_ID)

GSE59772_gene_names = assay(GSE59772) %>%
    as_tibble(rownames="Ensembl_Gene_ID") %>%
    pull(Ensembl_Gene_ID)


GSE10797 = makeObject(identifiers$GSE10797)
GSE10797_featuredata = rowData(GSE10797) %>%
    as_tibble(rownames = 'Ensembl_Gene_ID')

GSE10797_chr16 = GSE10797_featuredata %>%
    filter(Chromosome == '16') %>%
    pull(Ensembl_Gene_ID)

GSE10797_gene_names = assay(GSE10797) %>%
    as_tibble(rownames="Ensembl_Gene_ID") %>%
    pull(Ensembl_Gene_ID)


common_genes = c()
for (gene in GSE59772_gene_names) {
    if (gene %in% GSE10797_gene_names) {
        common_genes = c(common_genes, gene)
    }
}
```

Next, we filter the expression data to include only genes on chromosome 16, then filter again to only include genes the data sets have in common. We pivot and mutate to make the data ready for the boxplot. 
``` {r chr16_boxplot_2}
GSE10797_chr16_common = assay(GSE10797) %>%
    as_tibble(rownames = 'Ensembl_Gene_ID') %>%
    filter(Ensembl_Gene_ID %in% GSE10797_chr16) %>%
    filter(Ensembl_Gene_ID %in% common_genes) %>%
    pivot_longer(cols=-Ensembl_Gene_ID,names_to='Sample_ID',values_to='Expression_Level')%>%
    mutate(Dataset = 'GSE10797')

GSE59772_chr16_common = assay(GSE59772) %>%
    as_tibble(rownames = 'Ensembl_Gene_ID') %>%
    filter(Ensembl_Gene_ID %in% GSE59772_chr16) %>%
    filter(Ensembl_Gene_ID %in% common_genes) %>%
    pivot_longer(cols=-Ensembl_Gene_ID,names_to='Sample_ID',values_to='Expression_Level')%>%
    mutate(Dataset = 'GSE59772')
```

Finally, we combine the data sets using bind_rows() and read it into a ggplot. 
``` {r chr16_boxplot_3}
combined_data = bind_rows(GSE10797_chr16_common, GSE59772_chr16_common) %>%
    ggplot(aes(x = Dataset, y = Expression_Level, fill = Dataset)) +
    geom_boxplot() +
    theme_bw() +
    labs(x = 'Dataset', y='Gene Expression Level', title = 'Gene Expression Levels in Chromosome 16') 
combined_data
```

## Citation

Below is the citation output from using `citation('AnnotatedBCGEData')` in R. Please
run this yourself to check for any updates on how to cite __AnnotatedBCGEData__.

```{r 'citation', eval = requireNamespace('AnnotatedBCGEData')}
print(citation('AnnotatedBCGEData'), bibtex = TRUE)
```

Please note that the `AnnotatedBCGEData` was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://heidisteadman.github.io/RPracticePackage) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.
